components/Nav.tsx

```diff
@@ -25,10 +29,17 @@ const Nav: React.FC = (): React.JSX.Element => {
           GPRG
         </h1>
       </section>
-      <ul className="flex justify-between gap-3 md:gap-4 lg:gap-6 text-[0.75rem] md:text-[1rem]">
+      <span className="md:hidden flex items-center justify-center">
+          <Hamburger 
+            toggled={navbar}
+            toggle={() => setNavbar(!navbar)}
+            duration={0.5}/>
+      </span>
```

Comment:
for the toggle function, change it to () => setNavbar(prev => !prev).
This will help in rendering the up-to-date state of the hamburger menu.